### ISSUE TITLE
[Bug]: ModuleFederationPlugin remote not support IE11

### ISSUE THREAD

**Issue by wzc321** [web-infra-dev/rsbuild#2360]
### Version

```sh
System:
    OS: Windows 10 10.0.17134
    CPU: (4) x64 Intel(R) Core(TM) i5-7500 CPU @ 3.40GHz
    Memory: 4.01 GB / 15.93 GB
Browsers:
    Edge: Spartan (42.17134.1098.0)
    Internet Explorer: 11.0.17134.1
```

### Details

set browserslist:
"browserslist": [
"> 0.5%",
"not dead",
"IE 11"
]

set source.include 
`include: [/[\\/]node_modules[\\/]/],` or 
`include: [
      /[\\/]node_modules[\\/]@module-federation[\\/]/,
      { not: /[\\/]core-js[\\/]/ },
 ],`

and run `npm run build`  the remoteEntry.js are many `let` and `=>` but IE 11 not support

### Reproduce link

https://github.com/rspack-contrib/rspack-examples/tree/main/rsbuild/module-federation

### Reproduce Steps

1.Run npm install
2.Set package.json "browserslist": [ "> 0.5%", "not dead", "IE 11"   ]
3.Set source.include 
`include: [/[\\/]node_modules[\\/]/],` or 
`include: [
      /[\\/]node_modules[\\/]@module-federation[\\/]/,
      { not: /[\\/]core-js[\\/]/ },
 ],`
4.Run npm run build

**Comment by chenjiahan** [comment]
@ahabhgk The module federation plugin uses `data:application/node` to create virtual module and can not be matched by Rsbuild JS rules.

Can we change the `data:application/node` to `data:text/javascript` here?

https://github.com/web-infra-dev/rspack/blob/96d98de568856585c3e87be1508c3347e33a194a/packages/rspack/src/container/ModuleFederationPlugin.ts#L189-L190

**Comment by ahabhgk** [comment]
I think we can change it if all tests pass

**Comment by chenjiahan** [comment]
Get, I will try it

**Comment by wzc321** [comment]
Thanks for your reply, I do expect to fix this trouble in next release

**Comment by chenjiahan** [comment]
Fixed in [0.7.0-beta.6](https://github.com/web-infra-dev/rsbuild/releases/tag/v0.7.0-beta.6)
